### PR TITLE
Created a search bar for the map.

### LINKED ISSUE #65: Create (front-end) search for map markers.

### LINKED ISSUE #125: Figure out how to organize/hold the map markers in a data structure for the search bar.
Believe that the best implementation of a search bar would be using a leaflet plugin, since it seems relatively easy and customizable.

### LINKED ISSUE #126: Incorporate backend to make map marker search bar functional.

### LINKED ISSUE #127: Fix errors associated with station data when it is empty/undefined.

### PR DESCRIPTION
**Pull request by abijeet-dhillon**
Created a search bar for the map which allows users to quickly select different stations to display their data. This includes:

- **Functional Search Bar**
  The search bar allows users to start typing a station's name, and then displays a list of station names that match their inputs The user then selects the station name of the station they would like to view from the autocomplete list dropdown and then click the search button for the changes to take effect on the dashboard.

- **Error Handling for Null Data/Wrong Dates**
  Upon loading the page, if you don't see any data since it is not yet loaded on the database, the console will log a message saying that there is no data for this data. With your opinions, we could make it an issue to display an alert pop up notifying the user there isn't any data loaded for the selected date and inform them to choose another date.

Closes #65 #125 #126 #127

### CODE REVIEW

File: Site/bc_weather_station_dashboard/website/templates/weather.html [hunk 1]
```diff
@@ -131,9 +143,10 @@ <h2 class="card-text" id="wind-speed">N/A</h2>
                     </div>
                     <!-- Wind Direction Card -->
                     <div class="col-md-4">
-                        <div class="card rounded shadow mb-md-0 mb-4">
-                            <div class="card-body text-center" id="wind-direction">
```
Comment (robertyac) [inline Site/bc_weather_station_dashboard/website/templates/weather.html:135]: removing this messed up the wind direction widget

File: Site/bc_weather_station_dashboard/website/templates/weather.html [hunk 2]
```diff
@@ -131,9 +143,10 @@ <h2 class="card-text" id="wind-speed">N/A</h2>
                     </div>
                     <!-- Wind Direction Card -->
                     <div class="col-md-4">
-                        <div class="card rounded shadow mb-md-0 mb-4">
```
Comment (robertyac) [inline Site/bc_weather_station_dashboard/website/templates/weather.html]: Removing this got rid of the card styling

**Review by robertyac — DISMISSED** [review]
Search works well, great job!

There were a few lines you changed so I can't see the wind direction widget appearing anymore. After this is fixed should be good!

**Comment by carsondrobe** [comment]
I am getting this error:
`(index):107 Uncaught ReferenceError: searchBar is not defined
    at HTMLButtonElement.onclick ((index):107:97)`
    
it can't find the searchBar() function.

**Comment by carsondrobe** [comment]
> 

Nevermind it's working

File: Site/bc_weather_station_dashboard/website/templates/weather.html (hunk 1 above)
Comment (carsondrobe) [reply Site/bc_weather_station_dashboard/website/templates/weather.html:135]: I am seeing the same problem

**Review by carsondrobe — DISMISSED** [review]

File: Site/bc_weather_station_dashboard/website/templates/weather.html (hunk 1 above)
Comment (abijeet-dhillon) [reply Site/bc_weather_station_dashboard/website/templates/weather.html:135]: I readded the lines of code needed for wind direction widget, I don't even know how those got deleted... Good catch!

File: Site/bc_weather_station_dashboard/website/templates/weather.html (hunk 2 above)
Comment (abijeet-dhillon) [reply Site/bc_weather_station_dashboard/website/templates/weather.html]: Added it back.

**Review by robertyac — DISMISSED** [review]
Awesome, wind widget is back.

Are you planning on adding this to the fire page as well?

**Comment by abijeet-dhillon** [comment]
> Awesome, wind widget is back.
> 
> Are you planning on adding this to the fire page as well?

Yeah, just did, good catch again lol.

**Review by robertyac — COMMENTED** [review]
sorry imma come at you again but when you did that it works now but the styling changed on the fire page: 

Screenshot 2024-03-13 at 8 46 41 PM

**Comment by abijeet-dhillon** [comment]
> sorry imma come at you again but when you did that it works now but the styling changed on the fire page:
> 
> Screenshot 2024-03-13 at 8 46 41 PM

Try again now, should work.

**Review by robertyac — APPROVED** [review]
Awesome lgtm

**Review by Davis245 — APPROVED** [review]
Well done

**Pull request merged**